test/test_config.rb

```diff
@@ -58,4 +54,8 @@ def check_context(context, ssl: true)
       assert_equal(OpenSSL::SSL::VERIFY_NONE, context.ssl_options[:verify_ssl])
     end
   end
+
+  def config_file(name)
+    File.new(File.join(File.dirname(__FILE__), 'config', name))
+  end
```

Comment:
- please add a test for the backward compatibility you are maintaining. Then I'm good with this change. @simon3z please review, are we going to merge anything before the integration problem is fixed or should that be fixed first?